iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.h"
+
+#include "emitc/Dialect/EmitC/EmitCDialect.h"
+#include "iree/compiler/Dialect/VM/IR/VMOps.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/IR/Module.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+
+// Taken over from StandartToVM.
+// We need to replace the Op depending on the operand.
+// We could start with a conversion for IREE::VM::AddI32Op
+template <typename SrcOpTy, typename DstOpTy>
+class BinaryArithmeticOpConversion : public OpConversionPattern<SrcOpTy> {
+  using OpConversionPattern<SrcOpTy>::OpConversionPattern;
+
+ public:
+  BinaryArithmeticOpConversion(MLIRContext *context, StringRef funcName)
+      : OpConversionPattern<SrcOpTy>(context), funcName(funcName) {}
+
+ private:
+  LogicalResult matchAndRewrite(
+      SrcOpTy srcOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    typename SrcOpTy::Adaptor srcAdapter(operands);
+
+    MLIRContext *ctx = srcOp.getContext();
+
+    // name of the function to call
+    StringAttr callee_ = StringAttr::get(funcName, ctx);
```

Comment:
Remove trailing underscores in local variable names?

https://google.github.io/styleguide/cppguide.html#Variable_Names

Reply:
+1, but I think relevant style guide here is actually https://mlir.llvm.org/getting_started/DeveloperGuide/#style-guide

Reply:
Thanks for catching this. I removed the trailing underscores.

For variable names https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly advises:
"*Variable names should be nouns (as they represent state). The name should be camel case, and start with an upper case letter (e.g. Leader or Boats).*"

Till now, I think most variable names within `iree/compiler` start with a lower case letter...

Reply:
That's because of MLIR style, which uses camelBack (see my link).

I think the historical background is that MLIR started when LLVM was considering the camelBack proposal. it seemed like the proposal was going to go through, so they adopted it, but then it's kind of stagnated.

Reply:
Uhh, yeah. I shouldn't have stopped reading the MLIR style guide after the line "MLIR follows the LLVM style guide." :see_no_evil: Thanks for pointing me on that!